### DISCUSSION TITLE
[HELP] Question about tp.system.suggester

### DISCUSSION THREAD

**Discussion by FelipeRearden** [SilentVoid13/Templater#1011] [Help] [answered]
Hello friends :)

I have a question about a script that  I am trying to implement. 

### My Idea

I want to create a suggester to select a plugin from my vault and open the git repository of that plugin in safari to quickly create a new issue or discussion. 

The source of this script is from this discussion #707 where @welpdx kindly helped me to get the plugins list. At that time I was looking for a solution to open the plugins git from the Shortcuts App (it is working great !!!!) 

Now I want a solution when I am inside Obsidian ...

```
<%*
let content = ""
let url = ""
let plugins = Object.entries(this.app.plugins.plugins)
for (let i = 0; i < plugins.length; i++) {
let manifest = plugins[i][1]?.manifest
content += `"` + manifest?.name + `",` + " ";
url += `"` + manifest?.authorUrl + `",`  + " ";
}

var result = await tp.system.suggester([content], [url])
window.open("result");
%>
```

### My Problem 😭

📸 Modal
![1CAFA6A0-5CAE-4323-ADDD-C1943C33B56B](https://user-images.githubusercontent.com/85180696/215270767-b2058dbe-554a-4a15-9a38-da317e7b20e1.jpeg)

As you can see from the image above, all plugins names are been inserted into the first item of the suggester :(
- The suggester syntax is correct, but I dont know why is not creating the modal correctly :(
---

Does anybody could help me :) 

I dont have code skills and I didnt find any point to start here and at Discord. I am trying to solve this for hours but I dont have more fuel :(

Thank you very much for your help !!!!!

I wish you a fantastic day ☀️

**Comment by welpdx** [comment, marked as answer]
Hey @FelipeRearden! Its ya boi!

I see your code and you basically got it. What I would do different is I would create arrays instead of strings for `content` and `url`. The code below should work:

```js
<%*
let content = [];
let url = [];
let plugins = Object.entries(this.app.plugins.plugins)
for (let i = 0; i < plugins.length; i++) {
let manifest = plugins[i][1]?.manifest
content.push(manifest?.name)  
url.push(manifest?.authorUrl)
};

var result = await tp.system.suggester(content, url);
window.open(result);
%>
```




<details>
           <summary style="font-size:20">Demo</summary>
           <img src="https://i.imgur.com/e3lbWMK.gif" alt="demo" />
</details>


Hope that helps!

~welp

**Reply by FelipeRearden** [reply]
You are a legend @welpdx 🥇🥇🥇

Thank you very much for saving my Obsidian life one more time :)

it's working perfect, thanks for your help :)

I wish you a fantastic day ☀️

**Reply by welpdx** [reply]
Happy to help man! You have a good one too!
Tag me if you have another one

**Reply by FelipeRearden** [reply]
> Happy to help man! You have a good one too!

Thanks :)

> Tag me if you have another one

Thank you very much @welpdx !!!!!!!!

**Reply by FelipeRearden** [reply]
> Happy to help man! You have a good one too!
> Tag me if you have another one

Hello @welpdx !!!!

My good friend, I have another one for you when you have sometime :) 

→ https://github.com/SilentVoid13/Templater/discussions/1049

Thank you very  much for your help ☀️
